### DISCUSSION TITLE
Trouble getting full cp genome for Ericaceae species with abundant repeats

### DISCUSSION THREAD

**Discussion by Tmesipteris** [Kinggerm/GetOrganelle#256] [Q&A] [answered]
I am working on assembling the cp genome of a range of plant species and can easily obtain nicely assembled chloroplast genomes for those that are up to 160,000 bp in length using GetOrganelle. However, for two species in the Ericaceae family that have genomes between 170,000-200,000 bp long I cannot get the whole genome. So far I have tried using a -R of 100 to obtain more chloroplast reads, changing the -w value and increasing the disentangling time but I haven't changed -k values. However, no matter what I try I only get an incomplete genome consisting of scaffolds that represent around 135,000 bp in total.  
It doesn't seem to be due to a lack of chloroplast reads because read depths are over 400. and when I map the reads identified by GetOrganelle as cp reads (extended_1_paired.fq and  extended_2_paired.fq files) to the reference genome there is good coverage across the whole genome. However, when I align the scaffolds to the reference genome the two IRs are not completely represented suggesting that GetOrganelle is having trouble assembling the unusually long IRs in these species. Novoplasty also cannot assemble these long cp genomes.
Are there any ways to better assemble long chloroplast genomes using GetOrganelle?
Any help with this matter would be much appreciated.

**Comment by Kinggerm** [comment, marked as answer]
Sorry for the late reply.  As I searched, I noticed that you once posted #176 but closed it later. I don't know why I didn't get a chance to reply back then. Sorry about that.

First, when the result is incomplete/fragmented using current GetOrganelle versions, the `fasta` output is much less reliable than the ones when complete because 1) contig identification will be less accurate, 2) multiplicities will not be considered therefore repeats will not be considered. 

As for your trouble, the main trouble here is not the size of the target organelle genome, but the rampant repeats in notorious Ericaceae samples, causing the almost unsolvable tangled graph. Please visualize the `*.fastg` file (as indicated in the log file) to see the real trouble here. This problem fits the `answer 1.2` to [FAQ: What should I do with incomplete result/"broken assembly graph"?](https://github.com/Kinggerm/GetOrganelle/wiki/FAQ#what-should-i-do-with-incomplete-resultbroken-assembly-graph). 
So if I were you, I would either continue with the current incomplete result for downstream analysis (if it's mainly phylogenetic) or add long-read sequencing. I have another tool, [Traversome](https://github.com/JianjunJin/Traversome) that can combine GetOrganelle-generated graph with long-read to solve the problem and estimate potential isomers and frequencies.

**Reply by wen10-git** [reply]
Hello, @Kinggerm sorry to have disturbed you. I don't know if you are available.

Firstly I am not very familiar with the overall process. I noticed that based on these answers  (the section on long-reads) of yours, some of the methods in the literature (which I am going to possibly adopt) are loosely based on minimap2 mapping. It's not very detailed but I'm somewhat interested to know how you comment it?

"
**Assembly of chloroplast and mitochondrial genomes** 
The clean Illumina reads were used for draft assemblies of chloroplast and mitochondrial genomes with GetOrganelle[5] And preceding the filtered and corrected ONT reads were mapped on both organelle draft assemblies by minimap2 (version 2.11- 58 r797)[6]. All mapped reads were extracted for the following assemblies. Firstly, we used Canu (version 1.7) and SMARTdenovo (version 1.0.0) https://github.com/ruanjue/smartdenovo) to generate two primary assemblies. For the chloroplast genome, the assembly from SMARTdenovo was selected for high quality. Similarly, contigs from Canu (-correct -p assembly useGrid=true corOutCoverage=80 minReadLength=5000) were used to assemble the mitochondrial genome using SeqMan (version 11)8. Finally, we annotated and illustrated the two organelle genomes using the GeSeq web service (https://chlorobox.mpimpgolm.mpg.de/geseq.html).

Finally, the assembled mitochondrial genome gave a linear scaffold of 802,707 bp, with 45.87% average GC content. A total of 82 protein-coding genes were annotated in the genome, in addition to 60 annotated tRNA and 25 rRNA genes.
"

References: 
Main, Shuai Nie and others, Gapless genome assembly of azalea and multi-omics investigation into divergence between two species with distinct flower color, Horticulture Research, Volume 10, Issue 1, January 2023, uhac241, https://doi.org/10.1093/hr/uhac241;

(Delete Xiaoya Shi and others, The complete reference genome for grapevine (Vitis vinifera L.) genetics and breeding, Horticulture Research, Volume 10, Issue 5, May 2023, uhad061, https://doi.org/10.1093/hr/uhad061);

Correction and Reminder: Sorry, I made a mistake with the first reference at the beginning. It has been corrected now. Please take note! @Kinggerm 


Secondary, Yang, FS., Nie, S., Liu, H. et al. Chromosome-level genome assembly of a parent species of widely cultivated azaleas. Nat Commun 11, 5269 (2020). https://doi.org/10.1038/s41467-020-18771-4.



Thank you very much! Best regards!

Wen

**Reply by JianjunJin** [reply]
Assembling organelle genomes solely using long reads (like Shi et al. 2023 mentioned here; even using hifi reads see (Song et al. 2023)[https://www.frontiersin.org/articles/10.3389/fpls.2022.1065218/full]) will yield a much lower accuracy than short reads.

For a typical chloroplast genome (plastome) without annoying small repeats, the best solution concerning accuracy and efficiency so far is using GetOrganelle or similar assemblers to assemble plastome from short reads.

For repeat-rich plastomes or plant mitogenomes, the GetOrganelle-minimap-SMARTdenovo may not be the ideal solution due to poor accuracy and loss the information on potential isomers. A better approach should take advantage of both long and short reads. My ongoing project [Traversome](https://github.com/JianjunJin/Traversome) aims to provide the solution. However, I'm swamped by a few other projects and delayed Traversome for a while. Unfortunately, the current test stage is usable but not friendly to beginners.

**Reply by wen10-git** [reply]
I appreciate your response very much. Wish you good luck!

**Comment by Hua-CM** [comment]
Thank you for your reply. I have figured it out by adjusting the assembly graph using the Bandage manually.
